### PR TITLE
Updating circleci autobuild files and readme

### PR DESCRIPTION
**Pull request by exequielrafaela**
### Sync binbash jenkins-modules with head repo. 

<div align="middle">
  jenkins
</div>

### :notebook: Summary
Adding CircleCI auto-build job (triggered after merged PR only).  

### :man_technologist:  New Commits 
5c155b3 - Updating circleci autobuild files and readme

### :open_file_folder: Files changed
- `.circleci/config.yml` 
- `.gitignore`
- `Makefile`
- `README.md`

### :triangular_flag_on_post:  Post Tasks
- Validate auto-release CircleCi Job (https://circleci.com/gh/binbashar/jenkins-modules) passed and then manually create the release in github based on the automated tag. 

### :1st_place_medal:  TODO
- Add std doc-string and inline comments to every new module function.
- Test the new functions.
- Sync with latest LH module updates.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - run: git branch
+
+      - run:
+          name: release-patch-with-changelog-circleci
+          command: |
+            if git status | grep 'nothing to commit, working tree clean'; then
+              echo "==============================================================================================="
+              echo "release-patch-with-changelog-circleci"
+              echo "==============================================================================================="
+              git config --global user.email "info@binbash.com.ar"
```
Comment (diego-ojeda-binbash) [inline .circleci/config.yml]: @exequielrafaela I think you can set env vars in your repositories -- not sure if they are set per repository or if they can be made globally available). That way you store email and name there and then use them in your pipelines to improve that aspect just a little bit.

**Review by diego-ojeda-binbash — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (exequielrafaela) [reply .circleci/config.yml]: @diego-ojeda-binbash Thanks for taking the time for the review and great peace of advice. Do you mean setting ENV VARS in the  in the `./circleci/config.yml`? 

Something like: 
```
version: 2
jobs:
  build:
    docker:
      - image: smaant/lein-flyway:2.7.1-4.0.3
      - image: circleci/postgres:9.6-jessie
      # environment variables for all commands executed in the primary container
        environment:
          POSTGRES_USER: conductor
          POSTGRES_DB: conductor_test
```
source: https://circleci.com/docs/2.0/env-vars/

Comment (exequielrafaela) [reply .circleci/config.yml]: @diego-ojeda-binbash OJ I follow you now! Great point, this is actually possible per project (https://circleci.com/blog/devops-did-not-exist/) Sadly it's not cross-org, but at least every job that will be triggered inside the same project (1 project x repo) can use this ENV Vars from this CircleCI config. For example each 3 jobs for this same repo project will have centralized access to them:
1. **job:** sec-scan
2. **job:** build-test 
3. **job:** release

The main **con** is that is a manual UI config per project :-1: . Not sure if it's what we where expecting. However a big **pro** is that it works like a vault, keeping this ENV Vars protected from being access in plain text from the ./circleci/config.yml

image


What are your final thoughts about it?

Comment (diego-ojeda-binbash) [reply .circleci/config.yml]: @exequielrafaela Sounds like a step in the right direction. 👍

Comment (exequielrafaela) [reply .circleci/config.yml]: @diego-ojeda-binbash Great, I'll update the PR having the CircleCI env vars configured and validated before merging! Let's see if everything works as expected after the PR is actually merged :muscle:

**Pull request merged**